src/cmd.c

```diff
@@ -1089,6 +1090,139 @@ static void pctx(atci_param_t *param)
     request_confirmation = true;
 }
 
+static void cw(atci_param_t *param)
+{
+    uint32_t freq, timeout;
+    int32_t power;
+
+    if (!atci_param_get_uint(param, &freq)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_int(param, &power)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &timeout)) abort(ERR_PARAM);
+
+    log_debug("$CW: freq=%ld power=%ld timeout=%ld\r\n",  freq, power, timeout);
+
+    MlmeReq_t mlr = { .Type = MLME_TXCW };
+    mlr.Req.TxCw.Timeout = timeout;
+    mlr.Req.TxCw.Frequency = freq;
+    mlr.Req.TxCw.Power = power;
+
+    abort_on_error(LoRaMacMlmeRequest(&mlr));
+
+    OK_();
+}
+
+
+static void SX1276SetOpMode( uint8_t opMode )
+{
+    if( opMode == RF_OPMODE_SLEEP )
+    {
+        SX1276SetAntSwLowPower( true );
+    }
+    else
+    {
+        // Enable TCXO if operating mode different from SLEEP.
+        SX1276SetBoardTcxo( true );
+        SX1276SetAntSwLowPower( false );
+        SX1276SetAntSw( opMode );
+    }
+    SX1276Write( REG_OPMODE, ( SX1276Read( REG_OPMODE ) & RF_OPMODE_MASK ) | opMode );
+}
+
+
+static void cm_clk_irq_handler( void* context )
+{
+    (void) context;
+
+    static uint32_t i = 0;
+    i++;
+    gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, (i % 2) == 0);
+
+    /*
+        static bool pin = true;
+
+        // toggle every second when datarate is 4800
+        if(i % 4800 == 0)
+        {
+            atci_printf("X");
+            pin = !pin;
+        }
+
+        gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, pin);
+
+    */
+}
+
+
+static void cm(atci_param_t *param)
+{
+    (void)param;
+
+    // Example preamble 5, datarate 4800
+    // AT$CM 868300000,250000,4800,5,-10,3
+    uint32_t freq, timeout, fdev, datarate, preamble;
+    int32_t power;
+
+    if (!atci_param_get_uint(param, &freq)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &fdev)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &datarate)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &preamble)) abort(ERR_PARAM);
```

Comment:
Message, data, bitstream, payload, or something similar might be a better name for this parameter. Technically, this is not a preamble. The value that you specify here is the actual message.